python/lancedb/query.py

```diff
@@ -329,6 +334,30 @@ def to_arrow(self) -> pa.Table:
         )
         return self._table._execute_query(query)
 
+    def where(self, where: str, prefilter: bool = False) -> LanceVectorQueryBuilder:
```

Comment:
Why this one returns `LanceVectorQueryBuilder` while the other function returns `LanceQueryBUilder`?

Reply:
this overrides the superclass method since prefilter is only a thing for vector search right now